userland/init/Makefile

```diff
@@ -10,7 +10,7 @@ TARGET   = $(BIN_DIR)/init
 
 # Add `-no-pie` to disable the PIE feature that causes `gcc` to not create an
 # executable if needed, see: https://access.redhat.com/blogs/766093/posts/1975793
-CFLAGS = -Wl,-emain -Wall -pedantic -std=c11 -O2 -ffreestanding -nostdlib \
+CFLAGS = -Wl,-emain -Wall -pedantic -std=c11 -O0 -ffreestanding -nostdlib \
```

Comment:
Needed because adding `-O2` causes the program to crash with invalid opcode. I imagine it's because we need to do relocs/dyns.

Reply:
Very good finding, I confirm that `-O0` fixes an invalid opcode issue. I am curious, how did you find that?

Reply:
Sometimes compilers produce strange behaviour with optimizations, so changing to -O0 (no optimizations) made sense to me.

Reply:
ok, thanks